components/dashboards/org.wso2.carbon.dashboards.core/src/main/java/org/wso2/carbon/dashboards/core/DashboardDataProviderAuthorizer.java

```diff
@@ -245,16 +291,126 @@ private void assembleQuery(String username, DataProviderConfigRoot dataProviderC
     }
+
+    /**
+     * This method replaces the template values in the query with the values sent from front-end.
+     *
+     * @param username name of the logged in user
+     * @return id of the tenant
+     **/
+    private String getTenantId(String username) throws DataProviderException {
+        String adminServiceUrl;
+        String adminUsername;
+        String adminPassword;
+        try {
+            Map authConfigs = (Map) this.configProvider.getConfigurationObject(AUTH_CONFIGS_HEADER);
+            if (authConfigs == null) {
+                String error = "Cannot find " + AUTH_CONFIGS_HEADER + " in the deployment.yaml file.";
+                LOGGER.error(error);
+                throw new DataProviderException(error);
+            }
+            if (authConfigs.containsKey(AUTH_CONFIGS_PROPERTIES_HEADER)) {
+                Map properties = (Map) authConfigs.get(AUTH_CONFIGS_PROPERTIES_HEADER);
+                if (properties == null) {
+                    String error = AUTH_CONFIGS_PROPERTIES_HEADER + " header under " + AUTH_CONFIGS_HEADER + " in " +
+                            "the deployment.yaml file cannot be empty";
+                    LOGGER.error(error);
+                    throw new DataProviderException(error);
+                }
+                adminServiceUrl = getPropertyValueFromParentMap(properties, ADMIN_SERVICE_BASE_URL_KEY);
+                adminUsername = getPropertyValueFromParentMap(properties, ADMIN_SERVICE_USERNAME_KEY);
+                adminPassword = getPropertyValueFromParentMap(properties, ADMIN_SERVICE_PASSWORD_KEY);
+            } else {
+                String error = "Cannot find " + AUTH_CONFIGS_PROPERTIES_HEADER + " header under the "
+                        + AUTH_CONFIGS_HEADER + " in the deployment.yaml file.";
+                LOGGER.error(error);
+                throw new DataProviderException(error);
+            }
+        } catch (ConfigurationException e) {
+            String error = "Error occurred while getting the " + AUTH_CONFIGS_HEADER + " configuration from " +
+                    "deployment.yaml file.";
+            LOGGER.error(error);
+            throw new DataProviderException(error);
+        }
+        try {
+            Response response = DashboardAuthorizerServiceFactory.getAuthorizerHttpsClient(
+                    this.clientBuilderService, adminServiceUrl, adminUsername, adminPassword).getTenantId(username);
+            if (response == null) {
+                String error = "Response returned from the admin rest api is null.";
+                LOGGER.error(error);
+                throw new DataProviderException(error);
+            } else {
+                if (response.status() == 200) {
+                    TenantIdInfo tenantIdInfo = (TenantIdInfo) new GsonDecoder().decode(response, TenantIdInfo.class);
+                    String tenantId = tenantIdInfo.getTenantId().toString();
```

Comment:
Can there be instance where ```tenantIdInfo.getTenantId()```  becomes null?

Reply:
Fixed in 948aec8b481a3435bf89dab8232b2b1a23c0d374. (PS: This code segment is now moved to analytics-apim repo)